From 8ea22494471fd5d185e0e5545822313f844ace0c Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 28 Apr 2021 17:07:02 +0300
Subject: [PATCH] [SW-40350] fwupd: reuse the *_send_status() func

Reuse the module_fwupd_send_status() function also in
module_fwupd_setup() and module_fwupd_run() functions, so we have a
single send status entry point and no duplicated functionality.

Change-Id: I297f3b7708bf8656207a445817b2e9bddc11f92b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index ec68b55..9da7ba6 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -38,12 +38,20 @@
 
 static struct fwupd_state state;
 
+static void module_fwupd_send_status(enum hl_upd_sts sts, void *offset)
+{
+	struct hl_upd_status status;
+	uintptr_t off = (uintptr_t)offset;
+
+	status.val = 0;
+	status.status = sts;
+	status.offset = (u32)(off & ~HL_UPD_STS_OFFSET_MASK);
+	regmap_writel(status.val, mmUPD_STS);
+}
+
 void module_fwupd_setup(void)
 {
-	struct hl_upd_status status  = { .val = 0 };
-
-	status.status = HL_UPD_STS_RST_OK;
-	regmap_writel(status.val, mmUPD_STS);
+	module_fwupd_send_status(HL_UPD_STS_RST_OK, NULL);
 }
 
 enum cmd_check_res module_fwupd_check(void)
@@ -78,17 +86,6 @@
 	default:
 		return NOT_FOUND;
 	}
-}
-
-static void module_fwupd_send_status(enum hl_upd_sts sts, void *offset)
-{
-	struct hl_upd_status status;
-	uintptr_t off = (uintptr_t)offset;
-
-	status.val = 0;
-	status.status = sts;
-	status.offset = (u32)(off & ~HL_UPD_STS_OFFSET_MASK);
-	regmap_writel(status.val, mmUPD_STS);
 }
 
 static int module_fwupd_wfc(int timeout_ms)
@@ -354,7 +351,7 @@
 
 	switch (cmd.cmd) {
 	case HL_UPD_CLR_STS:
-		regmap_writel(HL_UPD_STS_NOOP, mmUPD_STS);
+		module_fwupd_send_status(HL_UPD_STS_NOOP, NULL);
 		preboot_puts_dbg("FW UPD CLR STS received\n");
 		break;
 	case HL_UPD_RST_STATE:
